### PR TITLE
Always invoke terraform get

### PR DESCRIPTION
**Pull request by aknuds1**
Always invoke [terraform get](https://www.terraform.io/docs/commands/get.html) in order to ensure that necessary modules are installed, to avoid build errors, rather than relying on whether the .terraform directory exists, since it might be stale.

See #571 for reference.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,16 +21,17 @@ installer-env: $(INSTALLER_BIN) terraformrc.example
...
-plan: installer-env $(BUILD_DIR)/.terraform
+plan: installer-env terraform-get
 	cd $(BUILD_DIR) && $(TF_CMD) plan $(TOP_DIR)/platforms/$(PLATFORM)
 
-apply: installer-env $(BUILD_DIR)/.terraform
+apply: installer-env terraform-get
 	cd $(BUILD_DIR) && $(TF_CMD) apply $(TOP_DIR)/platforms/$(PLATFORM)
 
-destroy: installer-env ${BUILD_DIR}/.terraform
+destroy: installer-env terraform-get
```
Comment (ggreer) [inline Makefile]: Are destroy, apply, & plan supposed to be phony? In other words: do you want them to run if a file named `destroy` is in the current dir?

Comment (aknuds1) [inline Makefile]: Will my change make `destroy`, `apply` and `plan` phony? I'm not a Make expert, but I figured that only the `terraform-get` target would become phony?

Comment (aknuds1) [inline Makefile]: I just tested creating such files FWIW (destroy/apply/plan), it seems the targets are executed no matter what, even if they exist. I would expect Make to skip them since corresponding files exist. This behaviour is the same without my patch.

Comment (ggreer) [inline Makefile]: It's always building because its dependencies also have missing outputs. I think your PR is better than what's currently in master, so I don't think it should be blocked on this.

I just didn't realize the Makefile was so hacky.

Comment (aknuds1) [inline Makefile]: Haha thanks for clarifying this @ggreer :) I guess these targets *should* be phony.

Comment (alexsomesan) [inline Makefile]: @aknuds1 

Can you please give a bit of background on why this change is needed?
What is your use-case where you need to run get before every apply?

The current behaviour is to invoke get only when module cache is missing.

Comment (aknuds1) [inline Makefile]: The case is when the cache is missing one or more modules. I don't know how it came about, but I imagine it's because modules were added in the source, so that the cache became stale. It's only logical that the cache can become out of sync with the source, we can't expect that it will stay relevant over time.

Comment (alexsomesan) [inline Makefile]: You're right that a new module being added would not trigger a get. In that sense, your change is brings more convenience. There is also a downside because this is invisible to the user and could "bite" unsuspecting users into applying changes that they weren't expecting. 

However, since the cache is nothing more than a bunch of symlinks to the actual module files, changes to those could also bite in the same way with the current state of things.

Things being the way they are, I think your change is useful and won't surprise with unexpected side-effects.

Please rebase and it should be good to merge.

**Comment by aknuds1** [comment]
Thanks @alexsomesan, I've rebased.

File: Makefile [hunk 2]
```diff
@@ -21,16 +21,17 @@ installer-env: $(INSTALLER_BIN) terraformrc.example
 localconfig:
 	mkdir -p $(BUILD_DIR)
 
-$(BUILD_DIR)/.terraform:
+.PHONY: terraform-get
```
Comment (alexsomesan) [inline Makefile]: One more thing: there already is a .PHONY section (at the end of the file).
Could you please move this over there?
It makes spotting target blocks easier when reading the Makefile.

**Comment by alexsomesan** [comment]
Thanks! I forgot one thing: see comment about the .PHONY location.

**Comment by aknuds1** [comment]
@alexsomesan Moved the phony declaration!

**Comment by alexsomesan** [comment]
ok to test
